### PR TITLE
Account Page WIP

### PR DESCRIPTION
**Pull request by jerminatorhits**
## Description
Why did you write this code?

## How to Test
- [ ] Visit XXX
- [ ] Click Y

Which devices did you test on?
- [ ] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Will this have future work?

## Learnings
Did you learn anything here you want to share with the team?

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -2,10 +2,12 @@ import * as React from 'react';
 import { Route, Switch } from 'react-router-dom';
 
 import Home from './home';
+import WorkAccount from './WorkAccount';
```
Comment (tc) [inline src/pages/index.tsx]: you can just call this `Account`; it's in a separate dir from the other components and it'll make it easier once it port it.

Comment (jerminatorhits) [reply src/pages/index.tsx]: 👍

File: src/pages/Account/AccountGeneral.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                  placeholder="First name"
+                  type="text" />
+                <FormFeedback>{errors.firstName}</FormFeedback>
+              </FormGroup>
+            </Col>
+            <Col md={6}>
+              <FormGroup inline>
+                <Label for={UserField.LAST_NAME} className="form-label">Last Name</Label>
+                <Field
+                  className={`form-control${errors.lastName ? ' is-invalid' : ''}`}
```
Comment (kcvan) [inline src/pages/Account/AccountGeneral.tsx]: we can use reactstrap `Input` and get pretty colors and stylings

File: src/pages/Account/Account.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                    <NavLink href="/work/account/verification">Verification</NavLink>
+                  </NavItem>
+                </Nav>
+                
+                <Container>
+                  <Row>
+                    <Col md={6}>
+                      <hr />
+                      <Switch>
+                        <Route exact path="/work/account/general" render={(props: RouterProps) => <AccountGeneral {...props} user={user} />} />
```
Comment (tc) [inline src/pages/Account/Account.tsx]: there's also https://github.com/reactjs/react-tabs for tabs

File: src/pages/Account/AccountGeneral.tsx (hunk 2 above)
Comment (jerminatorhits) [reply src/pages/Account/AccountGeneral.tsx]: swapped

File: src/pages/Account/Account.tsx (hunk 3 above)
Comment (jerminatorhits) [reply src/pages/Account/Account.tsx]: was able to 'reactstraps' tabs in an effort to maintain consistent styling.

File: src/pages/Account/AccountGeneral.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  [UserField.FIRST_NAME]: Yup.string()
+    .required('Please provide your first name.'),
+  [UserField.LAST_NAME]: Yup.string()
+    .required('Please provide your last name.'),
+  [UserField.EMAIL]: Yup.string(),
+  [UserField.ABOUT]: Yup.string()
+    .required('Please fill out the About section.'),
+})
+
+const defaultValues: FormValues = {
```
Comment (woeltjen) [inline src/pages/Account/AccountGeneral.tsx]: Would it be a useful simplification to do something like this?

    const defaultValues = Object.key(UserField).reduce(
      (values, key) => ({ ...values, [UserField[key]]: '' })
    );

That would reduce the number of places we'd need to update when adding a new form field, tho it only makes sense to do that if defaults will always be empty strings

Comment (jerminatorhits) [reply src/pages/Account/AccountGeneral.tsx]: good thing to consider, though I feel each field value declared explicitly feels a bit easier to read.

If other forms do include numbers/arrays/booleans, it would be easier to add / adjust at first glance.

**Comment by jerminatorhits** [comment]
closing to split into smaller PRs
